Foundation/NSOrderedSet.swift

```diff
@@ -46,85 +49,79 @@ open class NSOrderedSet : NSObject, NSCopying, NSMutableCopying, NSSecureCoding,
         }
     }
     
-    public required convenience init?(coder aDecoder: NSCoder) {
+    public required init?(coder aDecoder: NSCoder) {
         guard aDecoder.allowsKeyedCoding else {
             preconditionFailure("Unkeyed coding is unsupported.")
         }
+        _orderedStorage = NSMutableArray()
+        _storage = NSMutableSet()
         var idx = 0
-        var objects : [AnyObject] = []
         while aDecoder.containsValue(forKey: ("NS.object.\(idx)")) {
             guard let object = aDecoder.decodeObject(forKey: "NS.object.\(idx)") else {
                 return nil
             }
-            objects.append(object as! NSObject)
+            _orderedStorage.add(object)
+            _storage.add(object)
             idx += 1
         }
-        self.init(array: objects)
     }
     
     open var count: Int {
         return _storage.count
     }
 
     open func object(at idx: Int) -> Any {
-        return _SwiftValue.fetch(nonOptional: _orderedStorage[idx])
+        return _orderedStorage.object(at: idx)
     }
 
     open func index(of object: Any) -> Int {
-        return _orderedStorage.index(of: _SwiftValue.store(object)) ?? NSNotFound
+        return _orderedStorage.index(of: object)
     }
 
     public convenience override init() {
         self.init(objects: [], count: 0)
     }
 
-    public init(objects: UnsafePointer<AnyObject>!, count cnt: Int) {
-        _storage = Set<NSObject>()
-        _orderedStorage = [NSObject]()
+    public convenience init(objects: UnsafePointer<AnyObject>, count cnt: Int) {
```

Comment:
Inspecting the init in Xcode showed this specific variant as a convenience init while the one below was not, so I matched the behavior here as well

Reply:
If you follow the link for any of the methods in https://developer.apple.com/documentation/foundation/nsorderedset you will see the full function signature. It would be a good  idea to do a quick check of all of the methods and bring them into line with the Foundation documentation.

Reply:
Yep, I was checking this as I implemented them but missed this one. I will check again

Reply:
Found a couple more that need to be optional instead of implicitly unwrapped